ast/parser.go

```diff
@@ -897,7 +897,13 @@ func (p *Parser) parseSome() *Expr {
 			}
 
 			decl.Symbols = []*Term{term}
-			return NewExpr(decl).SetLocation(decl.Location)
+			expr := NewExpr(decl).SetLocation(decl.Location)
+			if p.s.tok == tokens.With {
+				if expr.With = p.parseWith(); expr.With == nil {
+					return nil
+				}
+			}
+			return expr
```

Comment:
So we should never continue processing, even if `p.s.tok != tokens.With`?

Reply:
The loop parsing all the expressions is somewhere outside of this; but for _this expression_, we're done. There's nothing that could come that would contribute to it.